lib/bolt/transport/ssh.rb

```diff
@@ -138,7 +144,15 @@ def run_task(target, task, arguments, options = {})
             end
 
             conn.with_remote_tempdir do |dir|
-              remote_task_path = conn.write_remote_executable(dir, executable)
+              if from_api?(task)
+                filename = task.name.split("::").last
+                remote_task_path = conn.write_executable_from_content(dir, task.file_content, filename)
+              else
+                executable = target.select_impl(task, PROVIDED_FEATURES)
+                raise "No suitable implementation of #{task.name} for #{target.name}" unless executable
```

Comment:
Before we had some notice earlier that the executable couldn't be identified. Although that doesn't look like it makes much difference in practice (it'll now happen after we try to establish a connection).